src/status_im/ui/screens/wallet/request/views.cljs

```diff
@@ -62,28 +62,33 @@
   (views/letsubs [{:keys [address]} [:popover/popover]
                   chain-id    [:chain-id]
                   width       (reagent/atom nil)]
-    [react/view
-     [react/view {:style {:padding-top 16 :padding-left 16 :padding-right 16}}
+    [react/view {:on-layout #(reset! width (-> % .-nativeEvent .-layout .-width))}
+     [react/view {:style {:padding-top 16 :padding-horizontal 16}}
       (when @width
-        [qr-code-viewer/qr-code-view (- @width 32) (eip681/generate-uri address {:chain-id chain-id})])
-      [react/text {:style {:font-size 13 :color colors/gray :margin-top 12 :margin-bottom 4}}
-       (i18n/label :t/ens-wallet-address)]
-      [react/view {:on-layout #(reset! width (-> % .-nativeEvent .-layout .-width))}
-       [react/text {:number-of-lines 1 :ellipsize-mode :middle
+        [qr-code-viewer/qr-code-view
+         (- @width 32)
+         (eip681/generate-uri address {:chain-id chain-id})])
+      [copyable-text/copyable-text-view
+       {:label           :t/ens-wallet-address
+        :container-style {:margin-top 12 :margin-bottom 4}
+        :copied-text     (eip55/address->checksum address)}
+       [react/text {:number-of-lines     1
+                    :ellipsize-mode      :middle
                     :accessibility-label :address-text
-                    :style           {:line-height 22 :font-size 15
-                                      :font-family "monospace"}}
+                    :style               {:line-height 22 :font-size 15
```

Comment:
I think we need not to provide line-height and font-size explicitly as it's the default typography. Actually, for `font-size` 15 we have `line-height` 21, but we shouldn't deviate from typography defaults. Not introduced in your PR anyways:) But maybe you can remove this explicit setting.

Reply:
I understand there is default typography spec implemented already. And there was an inquiry about overriding it in another bigger PR as well. The reason for the overriding of default typography implementation here is same as before - which is, it is specified in spec for share popover here https://www.figma.com/file/TNCyHKtR3sx5EL6YznFWUa4O/Profile?node-id=2796%3A22587.

In this one I did not leave a comment specifying reason for overriding, as in the other, because I'm beginning to suspect maybe there is a misunderstanding of what `typography spec` is for vs `spec for text elements in components`. @errorists please correct me if I'm wrong. As I understand it:
- Typography Spec - is for body text
- Text style spec for textual elements of components can override the Typography Spec if explicitly defined in spec for the component. There must've been a reason for overriding line-height in spec for popover or list-item.

Following those two rules still ensures `vertical rythm` which is the main concern behind having various `line-hight` spec.

Not following, or following only the first one can actually break `vertical rythm`. Because in the former there are padding/margin vertical centering etc to account for.

If this is true, better we update our collective mental-model about it.

Or maybe I'm taking figma spec too literally 😅 , believing everything specified there is not arbitrary, but have a reason like ensuring vertical rythm and such.

Reply:
Vertical rythm to me is how the negative space / white space flows along with visible elements to create kind of a rythm/beat. Also how in the srceenshot(iPhone Xs) below there is no clipping of content at the bottom. Even when it clips in devices of varying height and width it does so in a more or less sensible position.

![image](https://user-images.githubusercontent.com/42058840/64717832-bffa6180-d4e4-11e9-8c6f-dd8e0504735c.png)